app/Extension/Extension.php

```diff
@@ -0,0 +1,269 @@
+    protected $path;
+
+    /**
+     * Composer json of the package.
+     *
+     * @var array
+     */
+    protected $composerJson;
+
+    /**
+     * Whether the extension is installed.
+     *
+     * @var bool
+     */
+    protected $installed = true;
+
+    /**
+     * The installed version of the extension.
+     *
+     * @var string
+     */
+    protected $version;
+
+    /**
+     * Whether the extension is enabled.
+     *
+     * @var bool
+     */
+    protected $enabled = false;
+
+    /**
+     * @param       $path
+     * @param array $composerJson
+     */
+    public function __construct($path, $composerJson)
+    {
+        $this->path = $path;
+        $this->composerJson = $composerJson;
+        $this->assignId();
+    }
+
+    /**
+     * Assigns the id for the extension used globally.
+     */
+    protected function assignId()
+    {
+        list($vendor, $package) = explode('/', $this->name);
+        $package = str_replace(['mybb-ext-', 'mybb-'], '', $package);
+        $this->id = "$vendor-$package";
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __get($name)
+    {
+        return $this->composerJsonAttribute(Str::snake($name, '-'));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __isset($name)
+    {
+        return isset($this->{$name}) || $this->composerJsonAttribute(Str::snake($name, '-'));
+    }
+
+    /**
+     * Dot notation getter for composer.json attributes.
+     *
+     * @see https://laravel.com/docs/5.1/helpers#arrays
+     *
+     * @param $name
+     * @return mixed
+     */
+    public function composerJsonAttribute($name)
+    {
+        return Arr::get($this->composerJson, $name);
+    }
+
+    /**
+     * @param bool $installed
+     * @return Extension
+     */
+    public function setInstalled($installed)
+    {
+        $this->installed = $installed;
+
+        return $this;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isInstalled()
+    {
+        return $this->installed;
+    }
+
+    /**
+     * @param string $version
+     * @return Extension
+     */
+    public function setVersion($version)
+    {
+        $this->version = $version;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getVersion()
+    {
+        return $this->version;
+    }
+
+    /**
+     * Loads the icon information from the composer.json.
+     *
+     * @return array|null
+     */
+    public function getIcon()
+    {
+        if (($icon = $this->composerJsonAttribute('extra.mybb-extension.icon'))) {
+            if ($file = Arr::get($icon, 'image')) {
+                $file = $this->path.'/'.$file;
+
+                if (file_exists($file)) {
+                    $mimetype = pathinfo($file, PATHINFO_EXTENSION) === 'svg'
+                        ? 'image/svg+xml'
+                        : finfo_file(finfo_open(FILEINFO_MIME_TYPE), $file);
+                    $data = file_get_contents($file);
```

Comment:
We should probably do some more proper validation to make sure it's actually an image here.